flink-core/src/main/java/org/apache/flink/api/common/TaskInfo.java

```diff
@@ -132,11 +122,67 @@ public String getTaskNameWithSubtasks() {
 		return this.taskNameWithSubtasks;
 	}
 
+	/**
+	 * Returns the name of the task, appended with the subtask indicator and execution attempt id.
+	 *
+	 * @return The name of the task, with subtask indicator and execution id.
+	 */
+	public String getTaskNameWithSubtaskAndID() {
+		return taskNameWithSubtaskAndID;
+	}
+
 	/**
 	 * Returns the allocation id for where this task is executed.
 	 * @return the allocation id for where this task is executed.
 	 */
 	public String getAllocationIDAsString() {
 		return allocationIDAsString;
 	}
+
+	// ------------------------------------------------------------------------
+	//  Factory Utilities
+	// ------------------------------------------------------------------------
+
+	public static TaskInfo create(
+		String taskName,
+		int maxNumberOfParallelSubtasks,
+		int numberOfParallelSubtasks,
+		int subtaskIndex,
+		int attemptNumber,
+		String attemptIDAsString,
+		String allocationIDAsString) {
+
+		String taskNameWithSubtask = taskName + " (" + (subtaskIndex + 1) + '/' + numberOfParallelSubtasks + ')';
+		String taskNameWithSubtaskAndId = taskNameWithSubtask + " (" + attemptIDAsString + ')';
+
+		return new TaskInfo(
+			taskName,
+			taskNameWithSubtask,
+			taskNameWithSubtaskAndId,
+			maxNumberOfParallelSubtasks,
+			subtaskIndex,
+			numberOfParallelSubtasks,
+			attemptNumber,
+			allocationIDAsString);
+	}
+
+	@VisibleForTesting
+	public static TaskInfo create(
+		String taskName,
+		int maxNumberOfParallelSubtasks,
+		int subtaskIndex,
+		int numberOfParallelSubtasks,
+		int attemptNumber) {
+
+		String taskNameWithSubtask = taskName + " (" + (subtaskIndex + 1) + '/' + numberOfParallelSubtasks + ')';
+		return new TaskInfo(
+			taskName,
+			taskNameWithSubtask,
+			taskNameWithSubtask,
```

Comment:
this does not really fit. Why not creating a `TestingUtils#createTaskInfo` method which instantiates the `TaskInfo` with a random `ExecutionAttemptID`?

Reply:
Agree, then we could also avoid test related codes here.